bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/internal/SseItemStatesEventBuilder.java

```diff
@@ -110,71 +108,75 @@ public SseItemStatesEventBuilder(final BundleContext bundleContext, final @Refer
         return null;
     }
 
-    private @Nullable String getDisplayState(Item item, Locale locale) {
+    protected @Nullable String getDisplayState(Item item, Locale locale) {
         StateDescription stateDescription = item.getStateDescription(locale);
         State state = item.getState();
         String displayState = state.toString();
 
         if (!(state instanceof UnDefType)) {
             if (stateDescription != null) {
-                if (!stateDescription.getOptions().isEmpty()) {
+                String pattern = stateDescription.getPattern();
+                // If there's a pattern, first check if it's a transformation
+                if (pattern != null && TransformationHelper.isTransform(pattern)) {
+                    try {
+                        displayState = TransformationHelper.transform(pattern, state.toString());
+                    } catch (NoClassDefFoundError ex) {
+                        // TransformationHelper is optional dependency, so ignore if class not found
+                        // return state as it is without transformation
+                    } catch (TransformationException e) {
+                        logger.warn("Failed transforming the state '{}' on item '{}' with pattern '{}': {}", state,
+                                item.getName(), pattern, e.getMessage());
+                    }
```

Comment:
I don't understand this one. If `TransformationHelper` is optional and NOT present, wouldn't `TransformationHelper.isTransform(pattern)` also throw `NoClassDefFoundException` when invoked?

Reply:
@ghys Since this is you code, can you comment on this?

Reply:
My feeling is that TransformationHelper is always there.
What could lead to `NoClassDefFoundError` is probably the called transformation service itself, like the SCALE transformation service for example.
If I am right, this is just the comment that is not very precise.

Reply:
I see what you mean: the catch of NoClassDefFoundError  should be at a higher level to embed TransformationHelper.isTransform.
But why `TransformationHelper` would be optional ? It is part of the core framework and will always there, no ?

Reply:
@J-N-K : I just pushed a change that removes the catching of `NoClassDefFoundError`

Reply:
Let me know if you agree with this approach.

Reply:
@J-N-K : is it OK for you to no more consider `TransformationHelper` as optional ?